packages/react/src/hooks/usePrevious/usePrevious.ts

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useRef } from 'react'
+
+export const usePrevious = <T>(value: T): T => {
+  const ref = useRef<T>(value)
+
+  useEffect(() => {
+    ref.current = value
+  }, [value])
+
+  return ref.current
+}
```

Comment:
It is so COOL!

My previous review that return value will not change for refObj.current until hook is re-call is applied for this hook by so convenient and readable.

It is my code [sand-box link](https://codesandbox.io/s/misty-bash-hccjcm?file=/src/App.tsx) to understand usePrevious's purpose perfectly. 

It is my suggestion for you.

What about make hook for combining useState with usePrevious ?
My imagine API's interface is below

```tsx
const [state, setState, prevState] = useNewState(0)
```

The reason I suggested this is because I don't think `usePrevious` hook will be used often in itself, but if you want to check the previous value, assuming that the update is done with `useEffect` or `useState`, it will be used often!

Reply:
> const [state, setState, prevState] = useNewState(0)

After seeing your comment, I tried a lot to create an interface for that hook, but I think it was very difficult for me to name the hook. I also worry about whether Hook is playing too many roles. anyway I understood why you suggest it too.

If this hook is created, it is doubtful that it will be used in more cases than usePrevious. I wonder if there are any cases you are thinking of.

## New Suggestion: useIsChanged

After I saw your suggestion, I thought of a hook that can be used in more cases than `usePrevious`. I've named it `useIsChanged`. it looks like `ErrorBoundaryGroup`could use this hook as well.

```tsx
const useIsChanged(value: unknown){
  const previousValue = usePrevious(value)
  
  return previousValue !== value
}
```

### useIsChanged in ErrorBoundaryGroup
Isn't it more intuitive?

```tsx
export const ErrorBoundaryGroup = ({
  blockOutside = false,
  children,
}: {
  /**
   * If you use blockOutside as true, ErrorBoundaryGroup will protect multiple ErrorBoundaries as its children from external ErrorBoundaryGroup's resetKey
   * @default false
   */
  blockOutside?: boolean
  /**
   * Use multiple ErrorBoundaries inside of children
   */
  children?: ReactNode
}) => {
  const [resetKey, reset] = useKey()
  const parentGroup = useContext(ErrorBoundaryGroupContext)
  const isChangedParentGroupResetKey = useIsChanged(parentGroup?.resetkey)

  useEffect(() => {
    if (!blockOutside && isChangedParentGroupResetKey) {
      reset()
    }
  }, [isChangedParentGroupResetKey, reset, blockOutside])

  const value = useMemo(() => ({ reset, resetKey }), [reset, resetKey])

  return <ErrorBoundaryGroupContext.Provider value={value}>{children}</ErrorBoundaryGroupContext.Provider>
}
```

Reply:
> After seeing your comment, I tried a lot to create an interface for that hook, but I think it was very difficult for me to name the hook.

I deeply sympathize with what you say.
My suggestion hook is not only do TOO many things but also difficult intuitive naming.

> If this hook is created, it is doubtful that it will be used in more cases than usePrevious. I wonder if there are any cases you are thinking of.

As you understand my suggestion, my suggestion is for user can combining `useState` hook with `usePrevious`
Here is my example case.

1. I want to use state.
2. I want use previous state not using setter by callback function's args
3. so I wrote two lines of code (using `useState` and `usePrevious`)

Here is my sample code

[code-sandbox](https://codesandbox.io/s/misty-bash-hccjcm?file=/src/App.tsx)

Such example is common case of like logs.
But I recognize it is not common case,,, so I understand what you're worried about.

So, my suggestion for combining `useState` with `usePrevious` hook use case is not only common, but also can useless

Reply:
> New Suggestion: useIsChanged

I think it is COOL, intuitive, and good for DX!!

Reply:
> So, my suggestion for combining `useState` with `usePrevious` hook use case is not only common, but also can useless

Maybe you expect usePrevious's return value will be maintained as prev even though when component continuous rerendering. but, If component is rerendered, previousValue will be just previous Rendered value.

But I think it's good idea what you want. I want to imagine how to implement more it later.